File: src/main/kotlin/solutionsJava/Permutations.java
import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;
import java.util.Scanner;

public class Permutations {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int n = scanner.nextInt();

        List<Integer> result = permutations(n);
        if (!result.isEmpty()) {
            StringBuilder sb = new StringBuilder();
            for (int i = 0; i < result.size(); i++) {
                if (i > 0) sb.append(" ");
                sb.append(result.get(i));
            }
            System.out.println(sb);
        } else {
            System.out.println("NO SOLUTION");
        }
    }

    private static List<Integer> permutations(int n) {
        if (n == 1) return Arrays.asList(1);

        List<Integer> list = new ArrayList<>();
        if (n <= 3) return list;

        for (int i = 2; i <= n; i += 2) { list.add(i); }
        for (int i = 1; i <= n; i += 2) { list.add(i); }

        return list;
    }
}

File: src/main/kotlin/solutions/Permutations.kt
package uz.safix.solutions

import java.util.Scanner

fun main() {
    val scanner = Scanner(System.`in`)
    val n = scanner.nextInt()

    val result = permutations(n)
    if (result.isNotEmpty()) {
        println(result.joinToString(separator = " "))
    } else {
        println("NO SOLUTION")
    }
}

private fun permutations(n: Int): List<Int> = buildList {
    if (n == 1) return listOf(1)
    if (n <= 3) return@buildList

    for (i in 2..n step 2) { add(i) }
    for (i in 1..n step 2) { add(i) }
}